services/app-web/src/common-code/domain-models/submission.ts

```diff
@@ -80,9 +80,13 @@ const isDraftSubmission = (sub: unknown): sub is DraftSubmissionType => {
     return false
 }
 
+const naturalSort = (a: string, b: string): number => {
+    return a.localeCompare(b, "en", { numeric: true })
+}
+
 function submissionName(submission: SubmissionUnionType): string {
     const padNumber = submission.stateNumber.toString().padStart(4, '0')
-    return `${submission.stateCode.toUpperCase()}-${submission.programIDs[0].toUpperCase()}-${padNumber}`
+    return `MCR-${submission.stateCode.toUpperCase()}-${submission.programIDs.sort(naturalSort).join('-').toUpperCase()}-${padNumber}`
```

Comment:
Could we make a shared helper for submission name. This is used also in `app-api/emailer/temlates.ts` for emails.  I am remembering there was still a TODO to share that code somehow.

Reply:
I think this could be the shared helper since it’s in the domain model. I think I added that? I’m not sure if it’s being called in the email templates though